docs/docs/how-to/installing-shadcn.md

```diff
@@ -0,0 +1,347 @@
+# ShadCN
+
+## Installing ShadCN
+
+[ShadCN](https://ui.shadcn.com/) is an open source component library. 
+
+To install ShadCN, there are a few things you need to keep in mind regarding the structure of a Redwood application:
+
+- Redwood is a true full-stack frontend framework. Meaning, there's a proper frontend (`web/`) and backend (`api/`) separation.
+- Redwood uses yarn workspaces to manage the mono repo and its dependencies.
+
+If you want to install a package to use in the frontend, you'll need to run:
+
+```sh
+yarn workspace web add <package>
+```
+
+You can refer to the [Manual Installation documentation on ShadCN](https://ui.shadcn.com/docs/installation/manual), but you'll need to adapt the instructions to the Redwood structure.
+
+### 1. Add [Tailwind CSS](https://tailwindcss.com/)
+
+```sh
+yarn rw setup ui tailwindcss
+```
+
+This command will install Tailwind CSS and set up the necessary configuration files.
+
+### 2. Add the dependencies
+
+```sh
+yarn workspace web add tailwindcss-animate class-variance-authority clsx tailwind-merge
+```
+
+### 3. Add the icon library
+
+If you're using the `default` style, install `lucide-react`:
+
+```sh
+yarn workspace web add lucide-react
+```
+
+If you're using the `new-york` style, install `@radix-ui/react-icons`:
+
+```sh
+yarn workspace web add @radix-ui/react-icons
+```
+
+### 4. Configure path aliases
+
+Vite doesn't understand TypeScript's path mappings out of the box. You'll need to install a plugin to help Vite understand the path mappings.
+
+```sh
+yarn workspace web add vite-tsconfig-paths
+```
+
+and then within your `vite.config.js` file:
+
+```js
+import tsconfigPaths from 'vite-tsconfig-paths';
+
+export default {
+  plugins: [tsconfigPaths()],
+}
+```
```

Comment:
We omit the default contents of the `vite.config.js` here. I think it's obvious enough that this plugin should be in addition to the existing redwood one. Do you think it would be more or less confusing to alter this code block to include the default contents?

Reply:
Good question! ... I can add the default contents

Reply:
Thanks for the guide. I'm getting this error after setting-up everything (running rw v8.2.0)

```
yarn rw dev
```

```
web | ✘ [ERROR] "vite-tsconfig-paths" resolved to an ESM file. ESM file cannot be loaded by `require`. See https://vitejs.dev/guide/troubleshooting.html#this-package-is-esm-only for more details. [plugin externalize-deps]
```

Reply:
```
// vite.config.js

import dns from 'dns'

import type { UserConfig } from 'vite'
import { defineConfig } from 'vite'
import tsconfigPaths from 'vite-tsconfig-paths'

import redwood from '@redwoodjs/vite'

// So that Vite will load on localhost instead of `127.0.0.1`.
// See: https://vitejs.dev/config/server-options.html#server-host.
dns.setDefaultResultOrder('verbatim')

const viteConfig: UserConfig = {
  plugins: [redwood(), tsconfigPaths()],
}

export default defineConfig(viteConfig)

```

Reply:
I am also getting the same error. Was this ever resolved @ahaywood & @starfish456??